datastore/service.go

```diff
@@ -68,19 +81,30 @@ func (srv Service) CreateResource(ctx context.Context, namespace models.Namespac
 func (srv Service) UpdateResource(ctx context.Context, namespace models.NamespaceSpec, resourceSpecs []models.ResourceSpec, obs progress.Observer) error {
 	runner := parallel.NewRunner(parallel.WithLimit(ConcurrentLimit), parallel.WithTicket(ConcurrentTicketPerSec))
 	for _, resourceSpec := range resourceSpecs {
-		currentSpec := resourceSpec
-		repo := srv.resourceRepoFactory.New(namespace, currentSpec.Datastore)
+		incomingSpec := resourceSpec
+		repo := srv.resourceRepoFactory.New(namespace, incomingSpec.Datastore)
 		runner.Add(func() (interface{}, error) {
```

Comment:
same comments here too and there is code duplication any reason why we are not merging these flows.

Reply:
On this, it has been similar since the beginning (can be checked [here](https://github.com/odpf/optimus/blob/54f0784be99bd128a197ff0a1e4395c598d35efd/datastore/service.go)). If we it's required to be merged, then we have to also consider [this part](https://github.com/odpf/optimus/blob/54f0784be99bd128a197ff0a1e4395c598d35efd/ext/datastore/bigquery/bigquery.go) of the implementation since they are similar. If we can somehow merge the latter, then it shouldn't be much of issue about the former. If you are okay with it, further changes will be required to accommodate it.

Reply:
this is updated as mentioned. regarding the issue mentioned around bigquery, it will be addressed [here](https://github.com/odpf/optimus/issues/397).